azurerm/internal/services/iottimeseriesinsights/tests/resource_iot_time_series_insights_access_policy_test.go

```diff
@@ -0,0 +1,177 @@
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func TestAccAzureRMIoTTimeSeriesInsightsAccessPolicy_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_iot_time_series_insights_access_policy", "test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMIoTTimeSeriesInsightsAccessPolicyDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMIoTTimeSeriesInsightsAccessPolicy_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMIoTTimeSeriesInsightsAccessPolicyExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func TestAccAzureRMIoTTimeSeriesInsightsAccessPolicy_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_iot_time_series_insights_access_policy", "test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMIoTTimeSeriesInsightsAccessPolicyDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMIoTTimeSeriesInsightsAccessPolicy_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMIoTTimeSeriesInsightsAccessPolicyExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+			{
+				Config: testAccAzureRMIoTTimeSeriesInsightsAccessPolicy_update(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMIoTTimeSeriesInsightsAccessPolicyExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+			{
+				Config: testAccAzureRMIoTTimeSeriesInsightsAccessPolicy_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMIoTTimeSeriesInsightsAccessPolicyExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func testCheckAzureRMIoTTimeSeriesInsightsAccessPolicyExists(name string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		client := acceptance.AzureProvider.Meta().(*clients.Client).IoTTimeSeriesInsights.AccessPoliciesClient
+		ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+		// Ensure we have enough information in state to look up in API
+		rs, ok := s.RootModule().Resources[name]
+		if !ok {
+			return fmt.Errorf("Not found: %s", name)
+		}
+
+		id, err := parse.TimeSeriesInsightsAccessPolicyID(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		resp, err := client.Get(ctx, id.ResourceGroup, id.EnvironmentName, id.Name)
+		if err != nil {
+			return fmt.Errorf("Bad: Get on TimeSeriesInsightsAccessPolicyClient: %+v", err)
+		}
+
+		if utils.ResponseWasNotFound(resp.Response) {
+			return fmt.Errorf("Bad: Time Series Insights Access Policy %q (resource group: %q) does not exist", id.Name, id.ResourceGroup)
+		}
```

Comment:
Could we please move this if statement should move into the above `if err != nil` statement? Because when the service returns `404`, the `err` will always be non-nil, this means this piece of code is actually unreachable.